packages/database/src/prisma-api.ts

```diff
@@ -394,6 +394,26 @@ export function getSdk(
         },
       })
 
+      // if this user has already purchased this product, then an additional
+      // purchase (even for only 1 seat) should be processed as a bulk
+      // purchase so that they can distribute the seat to someone else.
+      //
+      // if an existingPurchase is found, but no existingBulkCoupon is found,
+      // then the logic below will account for that and create a new bulk
+      // coupon for the requested quantity.
+      const existingPurchase = await ctx.prisma.purchase.findFirst({
+        where: {
+          productId,
+          userId,
+          status: 'Valid',
+        },
+      })
```

Comment:
determine if there is an existing purchase of this product which we factor in below to determine if this new purchase should be considered a _bulk_ purchase